### ISSUE TITLE
[pedestrian] wont take the most common path

### ISSUE THREAD

**Issue by farfromrefug** [valhalla/valhalla#3547]
I cant find a way to tell valhalla to take the right path in some case.
Let s take the example in the screenshot. Most people will take the "red" path i drawn by hand. It is a `track` with a very light slope and just a few steps in the middle.
If i set `step_penalty` to 0 it will take steps all the way (which is really not for everyone). If i set it higher then valhalla will go all the way to a driving street as shown in the screenshot.
I see that as a bug in the sense that it should go through the "red" path.
I am wondering if adding `use_roads` would fix it. 
It is easily reproducible by using `#loc=17,45.199310,5.730497`
<img width="1789" alt="Screenshot 2022-02-22 at 13 02 24" src="https://user-images.githubusercontent.com/655344/155128705-06be954f-91e2-4cc9-ada5-5519dfc63de7.png">

**Comment by nilsnolde** [comment]
yeah I see that's a little unbalanced for what you want. balancing the costing to satisfy every expectation is really hard, if not close to impossible.

though in this case it's an absolute penalty, not a factor, so it should be pretty easy to debug with a simple test case. I'd also expect that a lower value would get you what you want. btw, I don't think `use_roads` is available for pedestrian (maybe it should be?). could you turn that into a gurka test? I think that'd be the easiest way to see if smth is going awry.

**Comment by farfromrefug** [comment]
@nilsnolde i totally agree about not being able to match every expectation but indeed here it seems a bit too much of a difference.
`use_roads` is indeed not available and i wonder if it could help.
What is a gurka test ? :P

**Comment by nilsnolde** [comment]
it might help but it also adds another level of complexity into the pedestrian costing. IMHO factors like that are much harder to get right. penalties are absolute, bit easier, at least for me.

gurka is the framework to build ascii maps and turn those into routable tiles. makes it very easy and visual to test certain isolated scenarios and understand the impact of costing options better, e.g. here testing that `top_speed` is doing what it's supposed to: https://github.com/valhalla/valhalla/blob/master/test/gurka/test_top_speed.cc. usually you find all ingredients you could need when browsing through some other files in that directory. if you can stand one up it'll be much easier for us to play around based on that or understand what's going on.

**Comment by farfromrefug** [comment]
@nilsnolde thanks for explaining. I kind of understand how it works. But what i have no idea about is what ascii map to create to reproduce what i am seeing in my live example. And i think there is also the elevation which is happening there. Gurka tests support grade?

**Comment by nilsnolde** [comment]
To build those maps I usually try to come up with a minimum example that should reproduce the problem I’m seeing, eg three equidistant routes to get from A to B, one with a minimal steps, one with steps all the way, one with no steps. Or smth like that. Your problem might not be super easily reproducible and depend on interaction with other parameters, but once it’s set up it’s fairly easy to play around and set breakpoints in the code and so on. Important thing is to not change too much at once. 

Fair warning, you might find that it’s a complex interaction of parameters in which case there might not be much we can do. Costing is a wily beast as you know:) this could be one edge case. It could also be a legitimate problem in the costing approach though.

I’d also leave out slopes for now. Not sure if you can tell gurka easily to build with elevation support. But if the start and end point have the same elevation in all routes it shouldn’t impact the costing relatively I think.
